Documentation/preparing-devel-environment.md

```diff
@@ -0,0 +1,90 @@
+# Preparing development environment and first rkt build
+This is example configuration for fresh instalation of Ubuntu 16.04 GNOME.
+
+### Required software
+You have to install few software packages to correctly build:
+```sh
+$ sudo apt-get install libssl-dev dh-autoreconf libacl1-dev libtspi-dev build-essential
+```
+See also dependencies [page](https://github.com/coreos/rkt/blob/master/Documentation/dependencies.md).
+
+### We definitely recommend using ccache for kvm flavor developers
+Ccache can save a lot of your time. Once build kernel can be taken from cache, so you don't have to build it over and over. You can get and configure ccache in few easy steps:
+
+```sh
+$ sudo apt-get install ccache
+$ ccache --max-size=10G
+$ sudo ln -s /usr/bin/ccache /usr/local/bin/gcc
+$ sudo ln -s /usr/bin/ccache /usr/local/bin/g++
+```
+We set maximum cache size to 10GB. (Default values are too small to perform kernel compilation caching)
+
+### Installing Go Programming Language for single-user
+```
+$ cd ~/Downloads
+$ curl -O https://storage.googleapis.com/golang/go1.6.1.linux-amd64.tar.gz
+$ tar -xvf go1.6.1.linux-amd64.tar.gz
+$ mv go ~/.local
+```
+
+### Get rkt repo
+I like to keep all my git repos in ~/Repos so..:
+```sh
+$ mkdir ~/Repos
+$ cd ~/Repos
+$ git clone https://github.com/coreos/rkt.git
+$ mkdir -p ~/.local/gopath/src/github.com/coreos
+$ git -C ~/.local/gopath/src/github.com/coreos clone https://github.com/coreos/rkt.git
+$ ln -s ~/.local/gopath/src/github.com/coreos Repos
+```
+### Add GO variables into .bashrc file:
+```sh
+export PATH=~/.local/bin:~/.local/go/bin:$PATH
+export GOPATH=~/.local/gopath
+export GOROOT=~/.local/go
+```
+
+### Building rkt
+
+Now we are ready to autogen and configure command with relevant arguments, for example:
+```sh
+$ ./autogen.sh && ./configure --enable-functional-tests --with-stage1-flavors=kvm
+```
+Now build rkt by:
+```sh
+$ make V=2 -j
+```
+REMEMBER: If you want test changes of somebody else:
+```sh
+$ git checkout <branch>
+$ make clean
+$ ./autgen.sh && ./configure <proper arguments>
+```
+
+### Few useful commands
+Just build and run tests:
+```sh
+$ ./tests/build-and-run-tests.sh -f kvm
+```
+Run only functional tests after build:
+```sh
+$ make functional-check
+```
+Check only one test:
+```sh
+$ make functional-check GO_TEST_FUNC_ARGS='-run TEST_NAME_HERE'
+```
+See more in [tests readme](https://github.com/coreos/rkt/blob/master/tests/README.md) page.
+
+Simple usage of rkt container (run, exit, remove):
+```sh
+$ sudo ./build-rkt-*/bin/rkt run --insecure-options=image --interactive docker://busybox
```

Comment:
missing info how to exit from container without interactive option for different flavors